src/Storage/Storage.Management/StorageAccount/StorageAccountBaseCmdlet.cs

```diff
@@ -212,6 +215,28 @@ protected static Track2Models.Encryption ParseEncryption(bool storageEncryption
             return accountEncryption;
         }
 
+        public static CloudStorageAccount GetCloudStorageAccount(Track2.StorageAccountResource storageAccountResource)
+        {
+            Uri blobEndpoint = storageAccountResource.Data.PrimaryEndpoints.Blob != null ? new Uri(storageAccountResource.Data.PrimaryEndpoints.Blob) : null;
+            Uri queueEndpoint = storageAccountResource.Data.PrimaryEndpoints.Queue != null ? new Uri(storageAccountResource.Data.PrimaryEndpoints.Queue) : null;
+            Uri tableEndpoint = storageAccountResource.Data.PrimaryEndpoints.Table != null ? new Uri(storageAccountResource.Data.PrimaryEndpoints.Table) : null;
+            Uri fileEndpoint = storageAccountResource.Data.PrimaryEndpoints.File != null ? new Uri(storageAccountResource.Data.PrimaryEndpoints.File) : null;
+            string key = null;
+            if (storageAccountResource.GetKeys()?.Value?.Keys != null && storageAccountResource.GetKeys().Value.Keys.Count > 0)
```

Comment:
If can't get valid key, should throw exception.

Reply:
might can change to more detail: Could not fetch storage account keys to build storage account context.

Reply:
Done